mlflow/utils/validation.py

```diff
@@ -346,6 +346,22 @@ def _validate_model_name(model_name):
         raise MlflowException("Registered model name cannot be empty.", INVALID_PARAMETER_VALUE)
 
 
+_RUN_ARTIFACTS_URI_REGEX = re.compile(r"^.+/[0-9a-f]{32}/artifacts/?")
+
+
+def _validate_and_resolve_source(source):
+    from mlflow.store.artifact.runs_artifact_repo import RunsArtifactRepository
+
+    if RunsArtifactRepository.is_runs_uri(source):
+        return RunsArtifactRepository.get_underlying_uri(source)
+    else:
+        if not _RUN_ARTIFACTS_URI_REGEX.match(source):
+            raise MlflowException(
+                f"Invalid source URI: '{source}'. Source URI must contain '<run_id>/artifacts'.",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
```

Comment:
An MLflow Model version doesn't necessarily need to come from a run. It could be an S3 location containing an MLflow Model directory, an Azure Blob storage location containing an MLflow Model directory, etc.

Instead of validating on this regex, we should only throw if the source is a `file:/` URI, a Unix path, or a Windows path.